docs/dev/release.rst

```diff
@@ -27,106 +30,98 @@ be "vX.YrcN" (also see `PEP 440 <https://www.python.org/dev/peps/pep-0440/>`_).
-You can create a clean checkout as follows (requires
-`SSH setup <https://docs.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh>`_)::
+The automated release infrastructure has proven to be reliable for a good number
+of releases now, so we've been using this faster version as the default release
+procedure. The longer procedure we previously used is still available in the
+:ref:`release-old` section.
 
-    mkdir jdaviz_for_release
-    cd jdaviz_for_release
-    git clone git@github.com:spacetelescope/jdaviz.git .
-    git fetch origin --tags
+You can do a release from your fork directly without a clean code check-out.
 
 #. Ensure `CI on Actions for main <https://github.com/spacetelescope/jdaviz/actions/workflows/ci_workflows.yml?query=branch%3Amain>`_
    and `RTD build for latest <https://readthedocs.org/projects/jdaviz/builds/>`_
    are passing.
 
+#. Lock down the ``main`` branch of the repository by setting the
+   `branch protection <https://github.com/spacetelescope/jdaviz/settings/branches>`_
+   rule for ``main`` to some high number required to merge, so that more PRs don't
+   get merged while you're releasing.
+
+#. Create a new branch on your fork and make sure you have updated tags too. Note
+   that the "x" here should actually be the letter "x", whereas the upper case "X"
+   and "Y" should be replace by your major and minor version numbers::
+
+     git fetch upstream main
+     git fetch upstream --tags
+     git checkout upstream/main -b vX.Y.x
+
 #. Update the ``CHANGES.rst`` file to make sure that all the user-facing changes are listed,
    and update the release date from ``unreleased`` to current date in the ``yyyy-mm-dd`` format.
    Remove any empty subsections.
 
 #. Update the ``CITATION.cff`` file's ``date-released`` and ``version`` fields.
    If there are new contributors to the project, add them in the ``authors``
-   section. Do not forget to commit your changes from the last two steps::
-
-     git add CHANGES.rst
-     git add CITATION.cff
-     git commit -m "Preparing release vX.Y.Z"
-
-#. Remove any untracked files. (WARNING: This will
-   permanently remove any files that have not been previously committed, so
-   make sure that you don't need to keep any of these files.)
-   This step is not needed if you have a fresh code checkout, but does not hurt either::
+   section.
 
-     git clean -xdf
+#. Update .github/workflows/ci_workflows.yml to add specific version branch to the
+   push/pull request specs. The relevant entries should look like::
 
-#. Tag the version you are about to release and sign it (optional but it is a good practice).
-   Signing requires
-   `GPG setup <https://docs.github.com/en/github/authenticating-to-github/managing-commit-signature-verification/adding-a-new-gpg-key-to-your-github-account>`_::
+     on:
+       push:
+         branches:
+           - v3.1.x
+       pull_request:
+         branches:
+           - v3.1.x
```

Comment:
Does this support wildcards?  (Can we just change this permanently to `-v*`?)

Reply:
Yes you can but just don't name other branches that would also match by accident (e.g., `victory` or `vontrapp`).

Reply:
Ok, so then depending on the style of matching (regex or wildcards), maybe something like `-v?.?.x`

Reply:
According to [this](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet) I think `'v[0-9]+.[0-9]+.[0-9]+'` is probably the most robust/specific way to specify this.